### PR TITLE
Tor algebra

### PR DESCRIPTION
**Pull request by larswinther**
The documentation has been updated. Improved error handling for local rings: version 2.0 could give nonsensical results if presented by a local ring whose defining ideal was not in the square of the maximal ideal.

### CODE REVIEW

File: M2/Macaulay2/packages/TorAlgebra.m2 [hunk 1]
```diff
@@ -360,7 +364,7 @@ toralgdata = R -> (
 		num = n + l*t + t^2 - t^3;
 		)
 	    else (
-		if isHomogeneous R then (
+		if isHomogeneous ideal R then (
```
Comment (DanGrayson) [inline M2/Macaulay2/packages/TorAlgebra.m2:367]: Was this change necessary?  The two conditions are equivalent.  (Just asking, no need to change it back.)

**Comment by larswinther** [comment]
Hi Dan,
That’s what I thought, but look at
i1 : FF =  ZZ/2;
i2 : Q = FF[x,y,z]
o2 = Q
o2 : PolynomialRing
i3 : I = ideal "x2y2+y4+xy2z+x2z2+xz3+yz3+y3, x3y+xy3+x3z+x2yz+xy2z+y2z2+x3+xy2+y3+x2z+xz2+yz2, x4+xy3+x3z+x2yz+xy2z+y2z2+y3+x+z, y4z+x3z2+xy2z2+y3z2+y2z3+xz4+y3z+x2+xy+yz"
             2 2    4      2     2 2      3      3    3   3       3    3     2         2     2 2    3      2    3
o3 = ideal (x y  + y  + x*y z + x z  + x*z  + y*z  + y , x y + x*y  + x z + x y*z + x*y z + y z  + x  + x*y  + y
     -------------------------------------------------------------------------------------------------------------
        2       2      2   4      3    3     2         2     2 2    3           4     3 2      2 2    3 2    2 3
     + x z + x*z  + y*z , x  + x*y  + x z + x y*z + x*y z + y z  + y  + x + z, y z + x z  + x*y z  + y z  + y z  +
     -------------------------------------------------------------------------------------------------------------
        4    3     2
     x*z  + y z + x  + x*y + y*z)
o3 : Ideal of Q
i4 :
     isHomogeneous I
o4 = false
i5 : isHomogeneous (Q/I)
o5 = true
Best,
Lars

**Comment by DanGrayson** [comment]
Thanks, that's a bug, and I've fixed it in commit 5e7e59be003d99942ffdd2cf7d43f697363ecc7d on the branch "development".

**Comment by larswinther** [comment]
Great, do you want me to update TorAlgebra accordingly and create a new pull request?

**Comment by DanGrayson** [comment]
> Great, do you want me to update TorAlgebra accordingly and create a new pull request? 

No, no need.
